numpy/lib/function_base.py

```diff
@@ -1840,13 +1840,13 @@ def gradient(f, *varargs, **kwargs):
         return outvals
 
 
-def diff(a, n=1, axis=-1):
+def diff(a, n=1, axis=-1, prepend=None, append=None):
```

Comment:
I think these actually need to be `np._NoValue`, else something like this doesn't work:

```
>>> data = np.array([2, 3], dtype=object)
>>> np.diff(data, prepend=None)
array([None, 1], dtype=object)  # expected
array([1], dtype=object)  # actual
```